src/cli/update_interactive_command.zig

```diff
@@ -0,0 +1,1216 @@
+            if (!initial_draw) {
+                Output.up(total_lines + 2);
+            }
+            Output.clearToEnd();
+
+            if (reprint_menu) {
+                var count: usize = 0;
+                for (state.selected) |sel| {
+                    if (sel) count += 1;
+                }
+                Output.prettyln("<r><green>✓<r> Selected {d} package{s} to update", .{ count, if (count == 1) "" else "s" });
+            }
+        }
+
+        while (true) {
+            if (!initial_draw) {
+                Output.up(total_lines);
+                Output.clearToEnd();
+            }
+            initial_draw = false;
+            total_lines = 0;
+
+            var displayed_lines: usize = 0;
+
+            // Group by dependency type
+            var current_dep_type: ?[]const u8 = null;
+
+            for (state.packages, state.selected, 0..) |*pkg, selected, i| {
+                // Print dependency type header with column headers if changed
+                if (current_dep_type == null or !strings.eql(current_dep_type.?, pkg.dependency_type)) {
+                    if (displayed_lines > 0) {
+                        Output.print("\n", .{});
+                        displayed_lines += 1;
+                    }
+
+                    // Count selected packages in this dependency type
+                    var selected_count: usize = 0;
+                    for (state.packages, state.selected) |p, sel| {
+                        if (strings.eql(p.dependency_type, pkg.dependency_type) and sel) {
+                            selected_count += 1;
+                        }
+                    }
+
+                    // Print dependency type - bold if any selected
+                    Output.print("  ", .{});
+                    if (selected_count > 0) {
+                        Output.pretty("<r><b>{s} {d}<r>", .{ pkg.dependency_type, selected_count });
+                    } else {
+                        Output.pretty("<r>{s}<r>", .{pkg.dependency_type});
+                    }
+
+                    // Calculate padding to align column headers with values
+                    var j: usize = 0;
+                    // Calculate actual displayed text length including count if present
+                    const dep_type_text_len: usize = if (selected_count > 0)
+                        pkg.dependency_type.len + 1 + std.fmt.count("{d}", .{selected_count}) // +1 for space
+                    else
+                        pkg.dependency_type.len;
+
+                    // The padding should align with the first character of package names
+                    // Package names start at: "    " (4 spaces) + "□ " (2 chars) = 6 chars from left
+                    // Headers start at: "  " (2 spaces) + dep_type_text
+                    // So we need to pad: 4 (cursor/space) + 2 (checkbox+space) + max_name_len - dep_type_text_len
+                    const padding_to_current = 4 + 2 + state.max_name_len - dep_type_text_len;
```

Comment:
this needs to handle when package names are smaller than dependency type string:

```
{
   "peerDependencies": {
        "jquery": "3.7.0"
    }
}
```

selecting jquery will add a counter, but there won't be room for it